src/CoreWCF.Primitives/src/CoreWCF/Security/MessageSecurityOverHttp.cs

```diff
@@ -0,0 +1,437 @@
+            }
+        }
+
+        internal bool WasAlgorithmSuiteSet
+        {
+            get { return this.wasAlgorithmSuiteSet; }
+        }
+
+        protected virtual bool IsSecureConversationEnabled()
+        {
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public SecurityBindingElement CreateSecurityBindingElement(bool isSecureTransportMode, bool isReliableSession, MessageSecurityVersion version)
+        {
+            if (isReliableSession && !this.IsSecureConversationEnabled())
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.SecureConversationRequiredByReliableSession)));
+            }
+
+            SecurityBindingElement result;
+            SecurityBindingElement oneShotSecurity = null;
+
+            bool isKerberosSelected = false;
+            bool emitBspAttributes = true;
+            if (isSecureTransportMode)
+            {
+                switch (this.clientCredentialType)
+                {
+                    case MessageCredentialType.None:
+                        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ClientCredentialTypeMustBeSpecifiedForMixedMode)));
+                    case MessageCredentialType.UserName:
+                        oneShotSecurity = SecurityBindingElement.CreateUserNameOverTransportBindingElement();
+                        break;
+                    //case MessageCredentialType.Certificate:
+                    //    oneShotSecurity = SecurityBindingElement.CreateCertificateOverTransportBindingElement();
+                    //    break;
+                    //case MessageCredentialType.Windows:
+                    //    oneShotSecurity = SecurityBindingElement.CreateSspiNegotiationOverTransportBindingElement(true);
+                    //    break;
+                    //case MessageCredentialType.IssuedToken:
+                    //    oneShotSecurity = SecurityBindingElement.CreateIssuedTokenOverTransportBindingElement(IssuedSecurityTokenParameters.CreateInfoCardParameters(new SecurityStandardsManager(new WSSecurityTokenSerializer(emitBspAttributes)), this.algorithmSuite));
+                    //    break;
+                    default:
+                        Fx.Assert("unknown ClientCredentialType");
+                        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException());
+                }
+                if (this.IsSecureConversationEnabled())
+                {
+                    result = SecurityBindingElement.CreateSecureConversationBindingElement(oneShotSecurity, true);
+                }
+                else
+                {
+                    result = oneShotSecurity;
+                }
+            }
+            else
+            {
+                //TODO 
+                //if (negotiateServiceCredential)
```

Comment:
This case is the full MessageSecurity scenario as the transport isn't secured. Throw PlatformNotSupportedException.

Reply:
Can you also remove the commented out code too